stdlib/public/Reflection/TypeRefBuilder.cpp

```diff
@@ -329,48 +329,46 @@ TypeRefBuilder::getClosureContextInfo(RemoteRef<CaptureDescriptor> CD) {
 
 void
 TypeRefBuilder::dumpTypeRef(RemoteRef<char> MangledName,
-                            std::ostream &OS, bool printTypeName) {
+                            FILE *file, bool printTypeName) {
   auto DemangleTree = demangleTypeRef(MangledName);
   auto TypeName = nodeToString(DemangleTree);
-  OS << TypeName << '\n';
+  fprintf(file, "%s\n", TypeName.c_str());
   auto TR = swift::Demangle::decodeMangledType(*this, DemangleTree);
   if (!TR) {
     auto str = getTypeRefString(MangledName);
-    OS << "!!! Invalid typeref: "
-       << std::string(str.begin(), str.end())
-       << '\n';
+    fprintf(file, "!!! Invalid typeref: %s\n", std::string(str.begin(), str.end()).c_str());
     return;
   }
-  TR->dump(OS);
-  OS << '\n';
+  TR->dump(file);
+  fprintf(file, "\n");
 }
 
-void TypeRefBuilder::dumpFieldSection(std::ostream &OS) {
+void TypeRefBuilder::dumpFieldSection(FILE *file) {
   for (const auto &sections : ReflectionInfos) {
     for (auto descriptor : sections.Field) {
       auto TypeDemangling =
         demangleTypeRef(readTypeRef(descriptor, descriptor->MangledTypeName));
       auto TypeName = nodeToString(TypeDemangling);
-      OS << TypeName << '\n';
+      fprintf(file, "%s\n", TypeName.c_str());
       for (size_t i = 0; i < TypeName.size(); ++i)
-        OS << '-';
-      OS << '\n';
+        fprintf(file, "-");
+      fprintf(file, "\n");
       for (auto &fieldRef : *descriptor.getLocalBuffer()) {
         auto field = descriptor.getField(fieldRef);
         auto fieldName = getTypeRefString(readTypeRef(field, field->FieldName));
-        OS << std::string(fieldName.begin(), fieldName.end());
+        fprintf(file, "%s", std::string(fieldName.begin(), fieldName.end()).c_str());
```

Comment:
@jckarter Is this necessary or is the field name going to be null-terminated?

Reply:
It would be better to use `fprintf(file, "%*s", (int)fieldName.size(), fieldName.data())`.

Reply:
Fixed ✅